editor/components/post-featured-image/index.js

```diff
@@ -38,15 +39,15 @@ function PostFeaturedImage( { featuredImageId, onUpdateImage, onRemoveImage, med
 						modalClass="editor-post-featured-image__media-modal"
 						render={ ( { open } ) => (
 							<Button className="button-link editor-post-featured-image__preview" onClick={ open } >
-								{ media && !! media.data &&
+								{ media &&
 									<ResponsiveWrapper
-										naturalWidth={ media.data.media_details.width }
-										naturalHeight={ media.data.media_details.height }
+										naturalWidth={ media.media_details.width }
+										naturalHeight={ media.media_details.height }
 									>
-										<img src={ media.data.source_url } alt={ __( 'Featured image' ) } />
+										<img src={ media.source_url } alt={ __( 'Featured image' ) } />
 									</ResponsiveWrapper>
 								}
-								{ media && media.isLoading && <Spinner /> }
+								{ ! media && <Spinner /> }
```

Comment:
What if a media request returns 404 ? This is a concern with treating "empty" as equivalent to "request in progress".

Reply:
I think I'm fine with this right now, we can revisit once/if we introduce request state